### PR TITLE
Viewport Addon

### LINKED ISSUE #1624: Resizing the preview iframe
Is there an available method to style or resize the preview iframe?

I would like to preview my components at various "screen sizes" to test some responsive CSS rulesets. Right now, I have to go fullscreen and resize the browser or enable the device mode using Chrome/Safari devtools. I could see making the testing workflow for a scenario like this much easier if the navigation menu could remain visible and the preview iframe just changed in size.

I tried out the third-party addon [storybook-host](https://github.com/philcockfield/storybook-host), but it creates/styles a div inside the preview iframe, skipping the responsive CSS rulesets.

### PR DESCRIPTION
**Pull request by saponifi3d**
## Description
This is a fairly early PR of a viewport plugin which allows you to resize the iframe of storybook.

### Remaining Tasks
- [ ] Fix lint issues
- [ ] Add unit tests
- [ ] Add documentation and screenshots to readme.

### Possible Features
- Responsive hooks, basically add the ability to drag the ends of the iframe to change the size
- Improve interaction with the `iframe` rather than just grabbing the iframe from the DOM add listeners to the iframe to allow us to interact with it

## Screen Cap
viewport

## Issues
Fixes #1624

### CODE REVIEW

File: addons/viewport/src/components/Panel.jsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    this.state = {
+      viewport: defaultViewport,
+      isLandscape: false,
+    };
+
+    this.props.channel.on('addon:viewport:update', this.changeViewport);
+  }
+
+  componentDidMount() {
+    this.iframe = document.getElementById(storybookIframe);
```
Comment (shilman) [inline addons/viewport/src/components/Panel.jsx]: LOL 🤣

**Comment by UsulPro** [comment]
Really great addon! Thanks for doing this!

I would add "write readme" to "Remaining Tasks".

I didn't dig into a code yet but I wonder whether it will be compatible with upcoming multistory feature (#1394)? Although not sure that it should, just want to share this thought

**Comment by shilman** [comment]
@saponifi3d @ndelangen I'd like to merge this into a feature branch in the storybook repo and add @saponifi3d to the storybooks org so we can more easily collaborate on fixing this up. OK?

**Comment by ndelangen** [comment]
Welcome @saponifi3d ! 👍

**Comment by markoshust** [comment]
Where is the git repo for this? Downloaded module from npm at https://www.npmjs.com/package/@storybook/addon-viewport -- all cool, but I have possible bugfix.

**Comment by saponifi3d** [comment]
@markoshust it's part of the storybook monorepo, if you go into `addons/viewport` it'll be there :)

**Comment by ndelangen** [comment]
@markoshust contact me on slack if you need help getting set up.

There's also the [CONTRIBUTING.md](https://github.com/storybooks/storybook/blob/release/3.3/CONTRIBUTING.md) document.

We're currently working off the release/3.3 branch as @saponifi3d mentioned.

**Pull request merged**